src/exclusive_pokemon.c

```diff
@@ -49,3 +48,269 @@ void InitializeExclusivePokemon(void)
         gUnknown_203B498->Exclusives[counter] = gExclusivePokemon[counter].in_rrt;
     }
 }
+
+void sub_8097FA8(u8 param_1)
+{   
+    gUnknown_203B498->unk48[param_1 >> 5] |= 1 << ((param_1 & 0x1f));
+}
+
+void sub_8097FD0(u8 param_1)
+{   
+    gUnknown_203B498->unk3C[param_1 >> 5] |= 1 << ((param_1 & 0x1f));
+}
+
+void sub_8097FF8(void)
+{
+    s32 iVar1;
+    s32 index;
+    struct ExclusivePokemonData *ptr;
+
+    for(index = 0; index < 0x40; index++)
+    {
+        ptr = gUnknown_203B498;
+        if (index < 0) {
+            iVar1 = index + 0x1f;
+        }
+        else
+        {
+            iVar1 = index;
+        }
+        if ((ptr->unk48[iVar1 >> 5] & (1 << (index - ((iVar1 >> 5) * 0x20)))) != 0) {
+            ptr->unk3C[iVar1 >> 5] |= (1 << (index - ((iVar1 >> 5) * 0x20)));
+        }
+    }
+    sub_8098080();
+}
+
+void sub_8098044(u8 param_1)
+{
+    gUnknown_203B498->unk3C[param_1 >> 5] &= ~(1 << (((param_1 & 0x1f))));
+    gUnknown_203B498->unk48[param_1 >> 5] &= ~(1 << (((param_1 & 0x1f))));
+}
+
+void sub_8098080(void)
+{
+    s32 index;
+
+    for(index = 0; index < 0x3; index++)
+    {
+        gUnknown_203B498->unk48[index] = 0;
+    }
+}
+
+u32 sub_80980A4(void)
+{
+    u32 temp = gUnknown_203B498->unk0;
+    gUnknown_203B498->unk0 = 1;
+    return temp;
+}
+
+void sub_80980B4(s16 pokeID)
+{
+    s32 index;
+    s32 pokeID_s32;
+    s32 pokeID_s32_1;
+    struct ExclusivePokemonData *ptr;
+
+    pokeID_s32 = pokeID;
+    pokeID_s32_1 = pokeID_s32;
+    if (pokeID_s32 != MONSTER_DECOY)
+    {
+        if(pokeID_s32 != MONSTER_STATUE)
+        {
+            if(pokeID_s32 != MONSTER_RAYQUAZA_CUTSCENE) {
```

Comment:
Is it possible to combine these three nested ifs?

Reply:
Looks like it tries to optimize it to a subtraction if instead of each case:
https://decomp.me/scratch/zuPpI

Reply:
Ah, nvm then.